packages/application-shell/src/components/application-shell/application-shell.tsx

```diff
@@ -41,10 +46,24 @@ type TApplicationShellProps = {
   onRegisterErrorListeners?: (args: { dispatch: Dispatch }) => void;
   onMenuItemClick?: (event: SyntheticEvent<HTMLAnchorElement>) => void;
   disableRoutePermissionCheck?: boolean;
+  enableStrictMode?: boolean;
```

Comment:
Nit: wondering if we should be more explicit that is related to React.

```suggestion
  enableReactStrictMode?: boolean;
```